lib/iris/tests/test_constraints.py

```diff
@@ -68,6 +68,15 @@ def test_constraints(self):
         sub_list = self.slices.extract(constraint)
         self.assertEqual(len(sub_list), 70 * 6)
 
+    def test_coord_availability(self):
+        constraint = iris.Constraint(model_level_number=lambda x: True)
+        result = self.slices.extract(constraint)
+        self.assertTrue(result)
+
+        constraint = iris.Constraint(latitude=lambda x: False)
+        result = self.slices.extract(constraint)
+        self.assertFalse(result)
```

Comment:
@rcomer This is a negative test i.e., the `latitude` coordinate is not in any of the `self.slices` `CubeList`, so the `result` is empty

Reply:
Huh.  Now I'm confused.  Because the pattern is `lambda x: False`, I assumed it was the other way around.  I.e. I thought the constraint was asking for cubes that don't have it, but they all do.  Looking at the actual `stock.realistic_4d` cube, they clearly won't.

< Had a play >

The result is the same whether you say `lambda x: True` or `lambda x: False`? 🤯

Reply:
Hmm, no.  I think `lambda x: False` always gives an empty list.

Reply:
```
In [30]: slices.extract(iris.Constraint(grid_latitude=lambda x: False))
Out[30]: []
```